### PR TITLE
reformat the python codes to pass flake8 test and change the way of import modules in Examples

### PR DESCRIPTION
**Pull request by shenmidelin**
I have used autopep8 to reformat the python codes, and now they pass the flake8 test. All the examples have been modified to use the new way of importing modules. A few bugs have been fixed. Readme is also updated. The new version of the CPC paper is now based on this version of the codes.

### CODE REVIEW

**Comment by mpmdean** [comment]
Hi @shenmidelin 

For the I/O module, I am unsure why we don't use some sort of off-the-shelf solution? Presumably there is something out there?

Best,
Mark

**Comment by shenmidelin** [comment]
@mpmdean , Sure, there are some available IO modules we can use, for example, when exchanging data, we can just use numpy.savetxt or hdf5. The functions in iostream.py are mostly used for temporary outputs for debugging, I think they are enough.

File: edrixs/coulomb_utensor.py [hunk 1]
```diff
@@ -529,129 +569,136 @@ def get_umat_kanamori_ge(norbs, U1, U2, J, Jx, Jp):
...
-               'dp':  get_umat_slater_dp,
-             't2gp':  get_umat_slater_t2gp,
-               'dd':  get_umat_slater_dd,
-               'fs':  get_umat_slater_fs,
-               'fp':  get_umat_slater_fp,
-               'fd':  get_umat_slater_fd,
-               'ff':  get_umat_slater_ff
-                }
-               
+        print("error in get_F0():  Unknown case name:", case)
```
[the remark this answers is not in the document]
Comment (shenmidelin) [reply edrixs/coulomb_utensor.py]: Thanks ! That will be a good idea. We can change those to raise Exception

File: edrixs/wannier_ham.py [hunk 2]
```diff
@@ -222,10 +223,10 @@ def kvec_from_file(self,fname):
         except IOError:
             print("File" + "\"" + fname + "\"" + "doesn't exists!")
```
[the remark this answers is not in the document]
Comment (shenmidelin) [reply edrixs/wannier_ham.py]: Yes, that line can be removed.

File: examples/more/RIXS/URu2Si2/plot_rixs_cut.py [hunk 3]
```diff
@@ -1,66 +1,81 @@
...
 import numpy as np
-import matplotlib
 import matplotlib.pyplot as plt
-from matplotlib.ticker import MultipleLocator, FormatStrFormatter
+from matplotlib.ticker import MultipleLocator
 
-font = {'family' : 'Times New Roman' ,
-	    'weight' : 'normal' ,
-		'size'   : '25'}
+font = {'family': 'Times New Roman',
```
[the remark this answers is not in the document]
Comment (shenmidelin) [reply examples/more/RIXS/URu2Si2/plot_rixs_cut.py]: I would prefer to keep the current values, it doesn't matter which fonts we use in examples.

File: examples/cpc/single_atom/hellorixs.py [hunk 4]
```diff
@@ -2,78 +2,72 @@
...
-
-font = {'family'  : 'Times New Roman' ,
-         'weight' : 'medium' ,
-         'size'   : '18'}
-plt.rc('font',**font)
+from matplotlib.ticker import MultipleLocator
+
+import edrixs
+
+font = {'family': 'Times New Roman',
```
[the remark this answers is not in the document]
Comment (shenmidelin) [reply examples/cpc/single_atom/hellorixs.py]: Keep the current values.

File: examples/more/RIXS/Ba2YOsO6/5d_single_atom/plot_rixs_map.py [hunk 5]
```diff
@@ -1,42 +1,51 @@
...
 
-font = {'family' : 'Times New Roman' ,
-	    'weight' : 'normal' ,
-		'size'   : '25'}
+font = {'family': 'Times New Roman',
+        'weight': 'normal',
+        'size': '25'}
 
-plt.rc('font',**font)
+plt.rc('font', **font)
```
[the remark this answers is not in the document]
Comment (shenmidelin) [reply examples/more/RIXS/Ba2YOsO6/5d_single_atom/plot_rixs_map.py]: Keep the current values.

File: examples/more/RIXS/Ba2YOsO6/5d_single_atom/plot_rixs_cut.py [hunk 6]
```diff
@@ -1,58 +1,73 @@
...
 import numpy as np
-import matplotlib
 import matplotlib.pyplot as plt
-from matplotlib.ticker import MultipleLocator, FormatStrFormatter
+from matplotlib.ticker import MultipleLocator
 
-font = {'family' : 'Times New Roman' ,
-	    'weight' : 'normal' ,
-		'size'   : '25'}
+font = {'family': 'Times New Roman',
```
[the remark this answers is not in the document]
Comment (shenmidelin) [reply examples/more/RIXS/Ba2YOsO6/5d_single_atom/plot_rixs_cut.py]: Keep the current values.

File: examples/more/RIXS/Ba2YOsO6/t2g_single_atom/plot_rixs_cut.py [hunk 7]
```diff
@@ -1,60 +1,75 @@
...
 
-font = {'family' : 'Times New Roman' ,
-	    'weight' : 'normal' ,
-		'size'   : '25'}
+font = {'family': 'Times New Roman',
+        'weight': 'normal',
+        'size': '25'}
 
-plt.rc('font',**font)
+plt.rc('font', **font)
```
[the remark this answers is not in the document]
Comment (shenmidelin) [reply examples/more/RIXS/Ba2YOsO6/t2g_single_atom/plot_rixs_cut.py]: Keep the current values.

File: examples/more/RIXS/Ba2YOsO6/t2g_single_atom/plot_rixs_map.py [hunk 8]
```diff
@@ -1,42 +1,52 @@
...
 
-font = {'family' : 'Times New Roman' ,
-	    'weight' : 'normal' ,
-		'size'   : '25'}
+font = {'family': 'Times New Roman',
+        'weight': 'normal',
+        'size': '25'}
 
-plt.rc('font',**font)
+plt.rc('font', **font)
```
[the remark this answers is not in the document]
Comment (shenmidelin) [reply examples/more/RIXS/Ba2YOsO6/t2g_single_atom/plot_rixs_map.py]: Keep the current values.

File: examples/more/RIXS/Ba3InIr2O9/t2g_two_site_cluster/plot.py [hunk 9]
```diff
@@ -1,29 +1,38 @@
...
-plt.ylim([max(omega),min(omega)])
-plt.xlim([min(eloss),max(eloss)])
-cax=plt.imshow(spectrum, extent=[min(eloss), max(eloss), max(omega),min(omega)], origin='upper', aspect='auto',cmap='jet', interpolation='bicubic')
-plt.plot(eloss, [11.215]*len(eloss), '--', color='white')
+font = {'family': 'Times New Roman',
+        'weight': 'normal',
+        'size': '20'}
+
+
+plt.rc('font', **font)
```
[the remark this answers is not in the document]
Comment (shenmidelin) [reply examples/more/RIXS/Ba3InIr2O9/t2g_two_site_cluster/plot.py]: Keep the current values.

File: examples/more/RIXS/LaNiO3_thin/plot_rixs_map.py [hunk 10]
```diff
@@ -1,62 +1,81 @@
...
 
-font = {'family' : 'Times New Roman' ,
-	    'weight' : 'normal' ,
-		'size'   : '25'}
+font = {'family': 'Times New Roman',
+        'weight': 'normal',
+        'size': '25'}
 
-plt.rc('font',**font)
+plt.rc('font', **font)
```
[the remark this answers is not in the document]
Comment (shenmidelin) [reply examples/more/RIXS/LaNiO3_thin/plot_rixs_map.py]: Keep the current values.

File: examples/more/RIXS/Pyrochlore_227/Ir_d5/single_atom/rixs.py [hunk 11]
```diff
@@ -1,95 +1,106 @@
...
         f.write(line)
     f.close()
 
     # plot rixs map
-    font = {'family' : 'Times New Roman' ,
-    	    'weight' : 'normal' ,
-            'size'   : '15'}
-    plt.rc('font',**font)
-    ax=plt.subplot(1,1,1)
+    font = {'family': 'Times New Roman',
```
[the remark this answers is not in the document]
Comment (shenmidelin) [reply examples/more/RIXS/Pyrochlore_227/Ir_d5/single_atom/rixs.py]: Keep the current values.

File: examples/more/RIXS/Pyrochlore_227/Os_d3/single_atom/rixs.py [hunk 12]
```diff
@@ -1,97 +1,108 @@
...
     # plot rixs map
-    font = {'family' : 'Times New Roman' ,
-    	    'weight' : 'normal' ,
-            'size'   : '15'}
-    plt.rc('font',**font)
-    ax=plt.subplot(1,1,1)
+    font = {'family': 'Times New Roman',
+            'weight': 'normal',
+            'size': '15'}
+    plt.rc('font', **font)
```
[the remark this answers is not in the document]
Comment (shenmidelin) [reply examples/more/RIXS/Pyrochlore_227/Os_d3/single_atom/rixs.py]: Keep the current values.

[the remark this answers is not in the document]
Comment (shenmidelin) [reply examples/more/RIXS/Pyrochlore_227/Os_d3/single_atom/rixs.py]: Keep the current values.

File: edrixs/wannier_ham.py [hunk 13]
```diff
@@ -53,35 +52,36 @@ def from_file(fname='wannier90_hr.dat'):
...
-                            hr[i,k,j] = np.float64(hr_real) + np.float64(hr_imag) * 1j
+                                irpt0 = i
+                            hr[i, k, j] = np.float64(
+                                hr_real) + np.float64(hr_imag) * 1j
                 # construct the HR instance
                 return HR(nwann, nrpt, irpt0, rpts, deg_rpt, hr)
 
         except IOError:
-            print("File:" + "\"" + fname + "\"" +  " doesn't exist!")
+            print("File:" + "\"" + fname + "\"" + " doesn't exist!")
```
[the remark this answers is not in the document]
Comment (shenmidelin) [reply edrixs/wannier_ham.py]: Agree

File: edrixs/soc.py [hunk 14]
```diff
@@ -25,83 +28,83 @@ def atom_hsoc(case, soc):
...
+        hsoc[11, 11] = -2.0
+        hsoc[12, 12] = 3.0
+        hsoc[10, 13] = sqrt6
+        hsoc[13, 13] = -3.0
         return 0.5 * soc * hsoc
 
     else:
-        print("don't support SOC for this case: ", case)
-        return
+        print("error in atom_hsoc: DO NOT support SOC for this case: ", case)
```
[the remark this answers is not in the document]
Comment (shenmidelin) [reply edrixs/soc.py]: Agree

File: edrixs/basis_transform.py [hunk 15]
```diff
@@ -261,120 +275,126 @@ def tmat_r2cub_f(ispin=False):
...
+        t_c2j[11, 11] = np.sqrt(2.0 / 7.0)
+        t_c2j[10, 12] = np.sqrt(6.0 / 7.0)
+        t_c2j[13, 12] = np.sqrt(1.0 / 7.0)
+        t_c2j[12, 13] = 1.0
 
         return t_c2j
 
     else:
-        print("NOT Implemented for this angular momentrum !!!", l)
+        print("error in tmat_c2j: Have NOT implemented for this case: ", orb_l)
```
[the remark this answers is not in the document]
Comment (shenmidelin) [reply edrixs/basis_transform.py]: Agree

File: edrixs/basis_transform.py [hunk 16]
```diff
@@ -1,212 +1,226 @@
...
+        t_c2r[6, 5] = -cone / sqrt2
         # fy(3x2-y2) = i/sqrt(2) ( |3,-3> + |3,3> )
-        t_c2r[0, 6] =  ci/sqrt2
-        t_c2r[6, 6] =  ci/sqrt2
+        t_c2r[0, 6] = ci / sqrt2
+        t_c2r[6, 6] = ci / sqrt2
     else:
-        print("don't support tmat_c2r for this case: ", case)
-        return
+        print("error in tmat_c2r: Do NOT support tmat_c2r for this case: ",
```
[the remark this answers is not in the document]
Comment (shenmidelin) [reply edrixs/basis_transform.py]: Agree

File: edrixs/iostream.py [hunk 17]
```diff
@@ -160,25 +130,46 @@ def write_tensor_nonzeros(tensor, fname, tol=1E-6):
...
     elif ndim == 4:
-        write_nonzeros_tensor4(tensor, fname, tol)
+        write_tensor_4(tensor, fname, only_nonzeros=only_nonzeros, tol=tol,
+                       fmt_int=fmt_int, fmt_float=fmt_float)
+    elif ndim == 5:
+        write_tensor_5(tensor, fname, only_nonzeros=only_nonzeros, tol=tol,
+                       fmt_int=fmt_int, fmt_float=fmt_float)
     else:
-        print("NOT implemented!")
+        print("error in write_tensor: ndim >5, not implemented !")
```
[the remark this answers is not in the document]
Comment (shenmidelin) [reply edrixs/iostream.py]: Agree

File: edrixs/coulomb_utensor.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (shenmidelin) [reply edrixs/coulomb_utensor.py]: Agree

[the remark this answers is not in the document]
Comment (shenmidelin) [reply edrixs/coulomb_utensor.py]: I have committed the changes.

File: edrixs/iostream.py (hunk 17 above)
[the remark this answers is not in the document]
Comment (shenmidelin) [reply edrixs/iostream.py]: I have committed the changes.

File: edrixs/basis_transform.py (hunk 16 above)
[the remark this answers is not in the document]
Comment (shenmidelin) [reply edrixs/basis_transform.py]: I have committed the changes.

File: edrixs/basis_transform.py (hunk 15 above)
[the remark this answers is not in the document]
Comment (shenmidelin) [reply edrixs/basis_transform.py]: I have committed the changes.

File: edrixs/soc.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (shenmidelin) [reply edrixs/soc.py]: I have committed the changes.

File: edrixs/wannier_ham.py (hunk 13 above)
[the remark this answers is not in the document]
Comment (shenmidelin) [reply edrixs/wannier_ham.py]: I have committed the changes.

File: edrixs/wannier_ham.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (shenmidelin) [reply edrixs/wannier_ham.py]: I have committed the changes.

**Pull request merged**